src/components/SnapshotDetails/tabs/SnapshotPipelineRunsTab.tsx

```diff
@@ -2,40 +2,61 @@ import * as React from 'react';
 import { useParams } from 'react-router-dom';
 import { Bullseye, Spinner } from '@patternfly/react-core';
 import { FilterContextProvider } from '~/components/Filter/generic/FilterContext';
+import { getErrorMessage } from '~/components/ImportForm/error-utils';
+import { SnapshotLabels } from '~/consts/snapshots';
+import { useSnapshot } from '~/hooks/useSnapshots';
 import { RouterParams } from '~/routes/utils';
 import { useNamespace } from '~/shared/providers/Namespace';
 import { PipelineRunLabel } from '../../../consts/pipelinerun';
-import { usePipelineRuns } from '../../../hooks/usePipelineRuns';
-import { StatusBox } from '../../../shared/components/status-box/StatusBox';
+import { usePipelineRun, usePipelineRuns } from '../../../hooks/usePipelineRuns';
 import PipelineRunEmptyState from '../../PipelineRun/PipelineRunEmptyState';
 import SnapshotPipelineRunsList from './SnapshotPipelineRunsList';
 
 const SnapshotPipelineRunTab: React.FC = () => {
   const { snapshotName, applicationName } = useParams<RouterParams>();
   const namespace = useNamespace();
+
+  const [snapshot, snapshotLoaded, snapshotError] = useSnapshot(namespace, snapshotName);
+
+  // The build pipelinerun should be queried separately since it will be the only pipelinerun
+  // created before the snapshot
+  const buildPipelineName = React.useMemo(
+    () =>
+      snapshotLoaded && !snapshotError
+        ? snapshot?.metadata?.labels?.[SnapshotLabels.BUILD_PIPELINE_LABEL]
+        : undefined,
+    [snapshot, snapshotLoaded, snapshotError],
+  );
+
+  const [buildPipelineRun] = usePipelineRun(namespace, buildPipelineName);
+
   const [pipelineRuns, loaded, LoadError, getNextPage, nextPageProps] = usePipelineRuns(
     namespace,
     React.useMemo(
       () => ({
         selector: {
+          // Test PipelineRuns are always created after the Snapshot, and the Build PipelineRun is already fetched
+          filterByCreationTimestampAfter:
```

Comment:
I'd like to double check something: I'm testing in my namespace and before this patch, I was able to see two PLRs for this snapshot's PLRs tab:

![Image](https://github.com/user-attachments/assets/9e25dd53-c877-4f12-9fa7-e542785109b9)

and with this patch, I'm only seeing the one with build type only:

![Image](https://github.com/user-attachments/assets/9412bfaf-3f4c-4e48-a23b-8d9c4b089b89)

is it expected? 🤔